store/tikv/split_region.go

```diff
@@ -16,70 +16,180 @@ package tikv
-				zap.Stringer("at", splitKey))
-			return 0, nil
-		}
-		res, err := sender.SendReq(bo, req, loc.Region, readTimeoutShort)
+
+	sender := NewRegionRequestSender(s.regionCache, s.client)
+	resp, err := sender.SendReq(bo, req, batch.regionID, readTimeoutShort)
+
+	batchResp := singleBatchResp{resp: resp}
+	if err != nil {
+		batchResp.err = errors.Trace(err)
+		return batchResp
+	}
+	regionErr, err := resp.GetRegionError()
+	if err != nil {
+		batchResp.err = errors.Trace(err)
+		return batchResp
+	}
+	if regionErr != nil {
+		err := bo.Backoff(BoRegionMiss, errors.New(regionErr.String()))
 		if err != nil {
-			return 0, errors.Trace(err)
+			batchResp.err = errors.Trace(err)
+			return batchResp
 		}
-		regionErr, err := res.GetRegionError()
-		if err != nil {
-			return 0, errors.Trace(err)
+		resp, err = s.splitBatchRegionsReq(bo, batch.keys, scatter)
+		batchResp.resp = resp
+		batchResp.err = err
+		return batchResp
+	}
+
+	spResp := resp.Resp.(*kvrpcpb.SplitRegionResponse)
+	regions := spResp.GetRegions()
+	if len(regions) > 0 {
+		// Divide a region into n, one can not need to be scattered,
+		// so n-1 needs to be scattered to other storage servers.
+		spResp.Regions = regions[:len(regions)-1]
+	}
+	if !scatter {
+		if len(spResp.Regions) == 0 {
+			return batchResp
 		}
-		if regionErr != nil {
-			err := bo.Backoff(BoRegionMiss, errors.New(regionErr.String()))
-			if err != nil {
-				return 0, errors.Trace(err)
-			}
+		logutil.BgLogger().Info("batch split regions complete",
+			zap.Uint64("batch region ID", batch.regionID.id),
+			zap.Stringer("first at", kv.Key(batch.keys[0])),
+			zap.Stringer("first new region left", logutil.Hex(spResp.Regions[0])),
+			zap.Int("new region count", len(spResp.Regions)))
+		return batchResp
+	}
+
+	for i, r := range spResp.Regions {
+		err = s.scatterRegion(r.Id)
+		if err == nil {
+			logutil.BgLogger().Info("batch split regions, scatter a region complete",
+				zap.Uint64("batch region ID", batch.regionID.id),
+				zap.Stringer("at", kv.Key(batch.keys[i])),
+				zap.Stringer("new region left", logutil.Hex(r)))
 			continue
 		}
-		splitRegion := res.Resp.(*kvrpcpb.SplitRegionResponse)
-		logutil.BgLogger().Info("split region complete",
-			zap.Stringer("at", splitKey),
-			zap.Stringer("new region left", logutil.Hex(splitRegion.GetLeft())),
-			zap.Stringer("new region right", logutil.Hex(splitRegion.GetRight())))
-		left := splitRegion.GetLeft()
-		if left == nil {
-			return 0, nil
+
+		logutil.BgLogger().Info("batch split regions, scatter a region failed",
+			zap.Uint64("batch region ID", batch.regionID.id),
+			zap.Stringer("at", kv.Key(batch.keys[i])),
+			zap.Stringer("new region left", logutil.Hex(r)),
+			zap.Error(err))
+		if batchResp.err == nil {
+			batchResp.err = err
 		}
-		if scatter {
-			err = s.scatterRegion(left.Id)
-			if err != nil {
-				return 0, errors.Trace(err)
-			}
+	}
+	return batchResp
+}
+
+// SplitRegions splits regions contains splitKeys into some regions.
```

Comment:
The term `some regions` is confusing, how about "SplitRegions splits regions by splitKeys"?